src/encoded/metadata.py

```diff
@@ -0,0 +1,187 @@
+    In Fourfront, there is custom structure looking for what is referred to as 'accession_triples', which is essentially
+    a 3-tuple containing lists of accesions that are either experiment sets, experiments or files
+
+    In SMaHT, in order to preserve similar structure, we eliminate logic for the first two (ExpSet and Exp) presuming
+    we will want to use those slots later, and provide only the files slot for now.
+
+    Alternatively, can accept a GET request wherein all files from ExpSets matching search query params are included.
+    """
+    ignored(context)
+    # Process arguments
+    if request.content_type == 'application/json':
+        try:
+            post_params = request.json_body
+            accessions = post_params.get('accessions', [])
+            type_param = post_params.get('type')
+            sort_param = post_params.get('sort')
+            download_file_name = post_params.get('download_file_name')
+            include_extra_files = post_params.get('include_extra_files', False)
+        except json.JSONDecodeError:
+            return Response("Invalid JSON format", status=400)
+    elif request.content_type == 'application/x-www-form-urlencoded':
+        post_params = request.POST
+        accessions = parse_qs(post_params.get('accessions')).get('accessions', [])
+        type_param = post_params.get('type')
+        sort_param = post_params.get('sort')
+        download_file_name = post_params.get('download_file_name')
+        include_extra_files = post_params.get('include_extra_files', False)
+    else:
+        return Response("Unsupported media type", status=415)
+
+    # One of type param or accessions must be passed
+    if not type_param and 'accessions' not in post_params:
+        return Response("Invalid parameters", status=400)
+
+    # Process the data
+    download_file_name = post_params.get('download_file_name')
+    if download_file_name is None:
+        download_file_name = 'metadata_' + datetime.utcnow().strftime('%Y-%m-%d-%Hh-%Mm') + '.tsv'
+
+    # generate search
+    search_param = {}
+    if not type_param:
+        search_param['type'] = 'File'
+    else:
+        search_param['type'] = type_param
+    if accessions:
+        search_param['accession'] = accessions
+    if sort_param:
+        search_param['sort'] = sort_param
+    search_iter = get_iterable_search_results(request, param_lists=search_param)
+
+    # Helper to grab field values if we reach a terminal field ie: not dict or list
+    def descend_field(d, field_name):
+        fields = field_name.split('.')
+        for field in fields:
+            d = d.get(field)
+        if isinstance(d, dict) or isinstance(d, list):  # we did not get a terminal field
+            return None
+        return d
+
+    # process search iter
+    data_lines = []
+    for file in search_iter:
+        line = []
+        for _, tsv_descriptor in TSV_MAPPING.items():
+            field = descend_field(file, tsv_descriptor.field_name()[0])
```

Comment:
The intent here is in case we want to search multiple fields for a value, obviously would be unused right now, but will update so it could be used in the future